scriptmodules/supplementary/skyscraper.sh

```diff
@@ -0,0 +1,696 @@
+}
+
+function build_skyscraper() {
+    qmake
+    make
+    md_ret_require="$md_build/Skyscraper"
+}
+
+function install_skyscraper() {
+    md_ret_files=(
+    'Skyscraper'
+    'LICENSE'
+    'README.md'
+    )
+}
+
+# Purge all Skyscraper caches
+function _purge_skyscraper() {
+    local platform
+
+    while read platform; do
+        # Find any folders underneath the Local DB cache and clear them
+        _clear_platform_skyscraper "$platform"
+    done < <(find "$configdir/all/skyscraper/dbs" -maxdepth 1 -mindepth 1 -type d -exec basename {} \;)
+}
+
+function _clear_platform_skyscraper() {
+    local platform="$1"
+
+    [[ ! -d "$configdir/all/skyscraper/dbs/$platform" ]] && return
+
+    # Remove any folder underneath the platform, try to keep any customized files
+    pushd "$configdir/all/skyscraper/dbs/$platform"
+    find . -maxdepth 1 -mindepth 1 -type d -print0 -exec rm -fr {} \;
+    rm -f db.xml
+    popd
+
+}
+# Purge by cached platform
+function _purge_platform_skyscraper() {
+    local options=()
+    local system
+
+    while read system; do
+        # If there is no 'db.xml' file underneath the folder, skip it, it means folder is empty
+        [[ ! -f "$configdir/all/skyscraper/dbs/$system/db.xml" ]] && continue
+
+        # Get the size on disk of the system and show it in the select list
+        local size=$(du -sh  "$configdir/all/skyscraper/dbs/$system" | cut -f1)
+        options+=("$system" "$size" OFF)
+    done < <(find "$configdir/all/skyscraper/dbs" -maxdepth 1 -mindepth 1 -type d -exec basename {} \;)
+
+    # If not folders are found, show an info message instead of the selection list
+    if [[ ${#options[@]} -eq 0 ]] ; then
+        printMsgs "dialog" "Nothing to delete ! No cached systems found in \n$configdir/all/skyscraper/dbs "
+        return
+    fi
+
+    local cmd=(dialog --backtitle "$__backtitle" --radiolist "Select platform to purge" 20 60 12)
+    local platform=$("${cmd[@]}" "${options[@]}" 2>&1 >/dev/tty)
+
+    # Exit if no platform chosen
+    [[ -z "$platform" ]] && return
+
+    _clear_platform_skyscraper "$platform"
+}
+
+
+function _get_ver_skyscraper() {
+    [[ -f "$md_inst/Skyscraper" ]] && "$md_inst/Skyscraper" -h | grep 'Running Skyscraper'  | cut -d' '  -f 3 2>/dev/null
+}
+
+function _latest_ver_skyscraper() {
+    # Prefer `wget` to `curl`, sinc it's already installed and used by the RetroPie script
+    wget -qO- https://api.github.com/repos/muldjord/skyscraper/releases/latest | grep -m 1 tag_name | cut -d\" -f4
+}
+
+# List any non-empty system found in the ROM parent folder
+function _list_systems_skyscraper() {
+    find "$romdir"  -mindepth 2 -maxdepth 2 -type f -not -name '+*' | sed 's%/[^/]*$%%' | sort -u
+}
+
+function configure_skyscraper() {
+    # When removing the package, clean up the Local DB
+    if [[ "$md_mode" == "remove" ]]; then
+        _purge_skyscraper
+        return
+    fi
```

Comment:
the purge function should probably be a remove_ function

Reply:
Initially it was, but I thought to spare an extra function definition and add it to the `configure`'s remove call. But it's clearer if moved to a separate function - I added the `remove_` back.